src/olympia/addons/fields.py

```diff
@@ -291,6 +293,28 @@ def to_internal_value(self, data):
                     gettext('Unknown min app version specified')
                 )
 
+            # For creations, at this point we have a potentially incomplete
+            # ApplicationsVersions instance, which could be missing its min
+            # and/or max. In Version.from_upload() we'll fill in the gaps with
+            # the manifest data, but we can't do that here, so we only validate
+            # the range for Android if we had both a min and a max provided.
+            # ApplicationsVersions.save() will silently correct min/max before
+            # saving if it's part of the forbidden range for Android anyway.
+            if (
+                'min' in min_max
+                and 'max' in min_max
+                and apps_versions.application == amo.ANDROID.id
```

Comment:
nit, `version_range_contains_forbidden_compatibility` checks application so this condition is redundant (if we don't care about the overhead of an extra function call, which we generally don't in python)

Reply:
Yes. `version_range_contains_forbidden_compatibility()` is technically not Android-specific, through its current implementation is. I was worried that if it was modified later to add additional more generic checks, we'd raise the wrong error message here (since the message _is_ Android-specific)